timesketch/lib/analyzers/domains.py

```diff
@@ -0,0 +1,281 @@
+        """Strip www. from beginning of domain names."""
+        if domain.startswith('www.'):
+            return domain[4:]
+        return domain
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = (
+            '{"query": { "bool": { "should": [ '
+            '{ "exists" : { "field" : "url" }}, '
+            '{ "exists" : { "field" : "domain" }} ] } } }')
+
+        return_fields = ['domain', 'url', 'message', 'human_readable']
+
+        events = self.event_stream(
+            '', query_dsl=query, return_fields=return_fields)
+
+        domains = {}
+        domain_counter = collections.Counter()
+        tld_counter = collections.Counter()
+
+        for event in events:
+            domain = event.source.get('domain')
+
+            if not domain:
+                url = event.source.get('url')
+                if not url:
+                    continue
+                domain_parsed = urlparse(url)
+                domain_full = domain_parsed.netloc
+                domain, _, _ = domain_full.partition(':')
+                event.add_attributes({'domain': domain})
+
+            if not domain:
+                continue
+
+            domain_counter[domain] += 1
+            domains.setdefault(domain, [])
+            domains[domain].append(event)
+
+            tld = self._get_tld(domain)
+            tld_counter[tld] += 1
+
+        watched_domains_list = current_app.config[
+            'DOMAIN_ANALYZER_WATCHED_DOMAINS']
+        domain_threshold = current_app.config[
+            'DOMAIN_ANALYZER_WATCHED_DOMAINS_THRESHOLD']
+        watched_domains_list.extend([
+            self._strip_www(x) for x, _ in domain_counter.most_common(
+                domain_threshold)])
+        watched_domains_list.extend([
+            x for x, _ in tld_counter.most_common(domain_threshold)])
+        watched_domains_list.extend(self.WATCHED_DOMAINS_BASE_LIST)
+        watched_domains_list_temp = set(watched_domains_list)
+        watched_domains_list = []
+        for domain in watched_domains_list_temp:
+            if domain in self.domain_scoring_whitelist:
+                continue
+            if any(domain.endswith(x) for x in self.domain_scoring_whitelist):
+                continue
+
+            if not '.' in domain:
+                continue
+            watched_domains_list.append(domain)
+
+        watched_domains = {}
+        for domain in watched_domains_list:
+            minhash = self._get_minhash_from_domain(domain)
+            watched_domains[domain] = minhash
+
+        similar_domain_counter = 0
+        for domain, count in domain_counter.iteritems():
+            emojis_to_add = [emojis.SATELLITE]
+            tags_to_add = []
+
+            if count == 1:
+                text = 'Domain: only occurance of domain'
+            else:
+                text = 'Domain seen: {0:d} times'.format(count)
```

Comment:
This will make any non-phishy domain loose too much information in human_readable, e.g:

<img width="903" alt="screenshot 2018-12-29 at 23 10 11" src="https://user-images.githubusercontent.com/316362/50542508-3eb1a780-0bbf-11e9-9e70-604afdf2c0c3.png">

Can we add the domain or url here as well?

Reply:
agree....done